### PR TITLE
feat: doc

### PR DESCRIPTION
**Pull request by BiswaViraj**
- Added migration guide for @novu/react

### CODE REVIEW

File: inbox/react/migration-guide.mdx [hunk 1]
```diff
@@ -0,0 +1,435 @@
+---
+title: "Migration Guide"
+sidebarTitle: "Migration Guide"
+description: "This guide outlines the key differences between the `@novu/notification-center` package and the new `@novu/react` package. Follow the steps below to migrate your application to the latest version."
```
Comment (LetItRock) [inline inbox/react/migration-guide.mdx:4]:
```suggestion
description: "This guide outlines the key differences between the `@novu/notification-center` package and the new `@novu/react` package. Follow the steps below to migrate your application to the latest Inbox version."
```

File: inbox/react/migration-guide.mdx [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+      applicationIdentifier="YOUR_APP_ID"
+      subscriberId="YOUR_SUBSCRIBER_ID"
+    >
+      <Notifications />
+    </Inbox>
+  );
+}
+
+export default Novu;
+```
```
Comment (LetItRock) [inline inbox/react/migration-guide.mdx:34]: @BiswaViraj In all these sections I'm missing the comparison steps, like the old way of doing it with `NotificationCenter` and a new way with the `Inbox`.

File: inbox/react/migration-guide.mdx [hunk 3]
```diff
@@ -0,0 +1,435 @@
+---
+title: "Migration Guide"
+sidebarTitle: "Migration Guide"
+description: "This guide outlines the key differences between the `@novu/notification-center` package and the new `@novu/react` package. Follow the steps below to migrate your application to the latest version."
+---
+
+## Getting Started
```
Comment (LetItRock) [inline inbox/react/migration-guide.mdx:7]: The migration guide should include:
- intro section that will explain the purpose of the guide
- version summary: briefly introduce the new version and highlight the major changes; list breaking changes, v2 compatibility; mention new features, performance, and bundle improvements; 
- migration steps: installation section
- api changes: in this section list all that you have but with old/new way comparison
- styling changes

File: inbox/react/migration-guide.mdx [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+```
+
+## New hooks
+
+Learn more about the new hooks available in the `@novu/react` package [here](hooks).
+
+- useNotifications
+- useCounts
+- usePreferences
+- useNovu
```
Comment (LetItRock) [inline inbox/react/migration-guide.mdx:435]: Let's hide this for now.

File: inbox/react/migration-guide.mdx (hunk 3 above)
Comment (LetItRock) [reply inbox/react/migration-guide.mdx:7]: cc @SokratisVidros

File: inbox/react/migration-guide.mdx (hunk 2 above)
Comment (BiswaViraj) [reply inbox/react/migration-guide.mdx:34]: Will add the old version 👍🏼

File: mint.json [hunk 5]
```diff
@@ -125,7 +125,8 @@
         "inbox/react/components",
         "inbox/react/preference",
         "inbox/react/multiple-tabs",
-        "inbox/react/advanced-configuration"
+        "inbox/react/advanced-configuration",
+        "inbox/react/migration-guide"
```
Comment (LetItRock) [inline mint.json:129]: I feel that we should move the migration guide to the second page in the sidebar wdyt?

File: inbox/react/migration-guide.mdx [hunk 6]
```diff
@@ -0,0 +1,737 @@
...
+      applicationIdentifier="YOUR_APP_ID"
+      subscriberId="YOUR_SUBSCRIBER_ID"
+    >
+      <Notifications />
+    </Inbox>
+  );
+}
+
+export default Novu;
+```
```
Comment (LetItRock) [inline inbox/react/migration-guide.mdx:136]: ✏ **suggestion:** I think that the first section should compare the old NC with popover to the current Inbox, as it's the default usecase.
Like:
```
Old:
import {
  NovuProvider,
  PopoverNotificationCenter,
  NotificationBell,
} from "@novu/notification-center";

 <NovuProvider
    subscriberId={"SUBSCRIBER_ID"}
    applicationIdentifier={"APPLICATION_IDENTIFIER"}
  >
    <PopoverNotificationCenter>
      {({ unseenCount }) => <NotificationBell unseenCount={unseenCount} />}
    </PopoverNotificationCenter>
  </NovuProvider>


New:
import { Inbox } from "@novu/react";

 <Inbox
    applicationIdentifier="YOUR_APP_ID"
    subscriberId="YOUR_SUBSCRIBER_ID"
  />
```

Comment (LetItRock) [reply inbox/react/migration-guide.mdx:136]: Then show how to use it without the popover and later with custom bell

File: inbox/react/migration-guide.mdx [hunk 7]
```diff
@@ -0,0 +1,737 @@
...
+
+```tsx
+import { NovuProvider, NotificationCenter } from "@novu/notification-center";
+
+<NovuProvider
+  subscriberId={"SUBSCRIBER_ID"}
+  applicationIdentifier={"APPLICATION_IDENTIFIER"}
+>
+  <NotificationCenter colorScheme="dark" />
+</NovuProvider>;
```
Comment (LetItRock) [inline inbox/react/migration-guide.mdx:116]: ✏ **suggestion:** please wrap with an example component like you do for the Inbox below

File: inbox/react/migration-guide.mdx [hunk 8]
```diff
@@ -0,0 +1,737 @@
...
+export default Novu;
+```
+
+### onPrimaryActionClick and onSecondaryActionClick
+
+Handle primary and secondary actions within a notification explicitly.
+
+- Old Implementation
+
+```tsximport {
```
Comment (LetItRock) [inline inbox/react/migration-guide.mdx:209]:
```suggestion
```tsx
import {
```

File: inbox/react/migration-guide.mdx [hunk 9]
```diff
@@ -0,0 +1,737 @@
...
+<PopoverNotificationCenter colorScheme="dark">
+  {({ unseenCount }) => (
+    <CustomBell
+      unseenCount={unseenCount}
+      colorScheme="dark"
+      unseenBadgeBackgroundColor="#FFFFFF"
+      unseenBadgeColor="#FF0000"
+    />
+  )}
+</PopoverNotificationCenter>
```
Comment (LetItRock) [inline inbox/react/migration-guide.mdx:325]: ✏ **suggestion:** please show the full example with wrapping into the component

File: inbox/react/migration-guide.mdx [hunk 10]
```diff
@@ -0,0 +1,737 @@
...
+
+For advanced positioning and styling of the notifications popover, integrate third-party popover libraries such as Radix UI.
+
+- Old Implementation
+
+```tsx
+<PopoverNotificationCenter position="left-start" offset={20}>
+  {({ unseenCount }) => <NotificationBell unseenCount={unseenCount} />}
+</PopoverNotificationCenter>
+```
```
Comment (LetItRock) [inline inbox/react/migration-guide.mdx:357]: same

File: inbox/react/migration-guide.mdx [hunk 11]
```diff
@@ -0,0 +1,737 @@
...
+export default Novu;
+```
+
+## Custom Notification Item
+
+Customize the appearance and structure of individual notification items using the `renderNotification` prop.
+
+- Old Implementation
+
+```tsx<PopoverNotificationCenter
```
Comment (LetItRock) [inline inbox/react/migration-guide.mdx:396]:
```suggestion
```tsx
<PopoverNotificationCenter
```

File: inbox/react/migration-guide.mdx [hunk 12]
```diff
@@ -0,0 +1,737 @@
...
+
+Customize the overall look and feel of the notification components using the appearance prop, which supports both CSS objects and class names (including Tailwind CSS classes).
+
+- Old Implementation
+
+```tsx
+<div className="bell">
+  <NovuProvider
+    subscriberId={process.env.REACT_APP_SUB_ID}
+    applicationIdentifier={process.env.REACT_APP_APP_ID}
```
Comment (LetItRock) [inline inbox/react/migration-guide.mdx:469]: ✏ **suggestion:** Let's use everywhere the same example for these fields:
```
applicationIdentifier="YOUR_APP_ID"
subscriberId="YOUR_SUBSCRIBER_ID"
```

**Review by LetItRock — COMMENTED** [review]
thanks @BiswaViraj much much better! 🙌

File: mint.json (hunk 5 above)
Comment (BiswaViraj) [reply mint.json:129]: 🤔 Usually, I find the "migration guide" at the bottom as the migration section becomes obsolete after some time(like after a month or two) once almost all the users are on the new version.

**Review by LetItRock — APPROVED** [review]
awesome! 🙌

**Pull request merged**